daemon/containerd/image_pull.go

```diff
@@ -217,6 +217,18 @@ func (i *ImageService) pullTag(ctx context.Context, ref reference.Named, platfor
 			}
 			return errdefs.NotFound(fmt.Errorf("pull access denied for %s, repository does not exist or may require 'docker login'", reference.FamiliarName(ref)))
 		}
+		if cerrdefs.IsNotFound(err) {
+			// Transform "no match for platform in manifest" error returned by containerd into
+			// the same message as the graphdrivers backend.
+			// The one returned by containerd doesn't contain the platform and is much less informative.
+			if strings.Contains(err.Error(), "platform") {
```

Comment:
😢 on the string-matching. Guess we have no real other option here 😞 

Wondering if there would be options to do so through richer errors from the containerd code 🤔 (or perhaps the containerd code to return a better error-message)

Reply:
Went looking if that's something the matcher could return, but that's using a simple `bool` as interface, so no options there 😢 https://github.com/moby/moby/blob/5ea96cad4b72edb56c2f0169fee6a864126ca9c1/vendor/github.com/containerd/platforms/platforms.go#L133-L136